### PR TITLE
#139 [User Management] [Endpoint] Update a specific User

### LINKED ISSUE #139: [User Management] [Endpoint] Update an specific User
Related to https://github.com/pimcore/studio-ui-bundle/issues/180

### PR DESCRIPTION
**Pull request by martineiber**
## Changes in this pull request
Resolves #139

### CODE REVIEW

File: src/User/Service/UserUpdateService.php [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+        $this->user->setPermissions($permissionsToSet);
+    }
+
+    /**
+     * @throws NotFoundException
+     */
+    private function updateRoles(array $rolesToSet): void
+    {
+        $roles = array_map(function ($role) {
```
Comment (mattamon) [inline src/User/Service/UserUpdateService.php]: Can this array_map be static? :)

File: src/User/Service/UserUpdateService.php [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    }
+
+    /**
+     * @param UserWorkspace[] $assetWorkspacesToSet
+     *
+     * @throws ParseException
+     */
+    private function checkForDuplicateWorkspaces(array $assetWorkspacesToSet): void
+    {
+        $paths = array_map(function ($workspace) {
```
Comment (mattamon) [inline src/User/Service/UserUpdateService.php]: I would in general check the array_map for static

**Review by mattamon — CHANGES_REQUESTED** [review]
The user is a chunky boy, but overall LGTM.

Can you please check the array_maps? :)

**Comment by martineiber** [comment]
@mattamon Yeah.. the User is a big boy :D I tried to split it up as good as possible, but it is still really big :(

I added the static to all arrray_map(). Thx for the catch.

**Review by mattamon — APPROVED** [review]
🦭

**Pull request merged**